CustomDimensions.php

```diff
@@ -280,10 +281,13 @@ public function addConversionInformation(&$conversion, $visitInformation, Tracke
 
         foreach ($dimensions as $dimension) {
             $index = (int) $dimension['index'];
-            if ($dimension['scope'] === self::SCOPE_VISIT && in_array($index, $conversionIndexes)) {
+            if (in_array($index, $conversionIndexes)) {
                 $field = LogTable::buildCustomDimensionColumnName($dimension);
 
-                if (array_key_exists($field, $visitInformation)) {
+                if ($dimension['scope'] === self::SCOPE_ACTION && array_key_exists($field, $actionDimensions)) {
+                    $conversion[$field] = $actionDimensions[$field];
+                }
+                else if ($dimension['scope'] === self::SCOPE_VISIT  && array_key_exists($field, $visitInformation)) {
```

Comment:
Good point @tsteur 

> Because in log_conversion table we also have only 5 columns by default (custom_dimension1 to custom_dimension5) it is not clear to which custom dimension it actually references. It would need more like a custom_dimension_index_1 to custom_dimension_index_10 column. 

Maybe could be named `custom_dimension_1_visit` and `custom_dimension_1_action` to make it more clear and not introduce a new naming/indexing.

> Then we would need to change the archiver to generate reports for goal reports differently and some other logic likely in menu etc. This is quite some work.

Yes would need to have the archiver use these special columns from log_conversion. 
Not sure about changing menus and/or whether other changes would be needed?

> I think I would now recommend to change this, if we want, for Piwik 4. And I would likely recommend to @code-penguin to apply this logic rather in a new plugin for now to have this effect already.

Not sure re: Piwik 4 as it sounds lke the changes could maybe be done in CustomDimensions plugin only and so would be do-able independantly of Piwik 3/Piwik 4. Maybe @code-penguin you would like to try this in the plugin?